gdnative-core/src/core_types/string.rs

```diff
@@ -612,6 +612,69 @@ where
     }
 }
 
+#[cfg(feature = "serde")]
+mod serde {
+    use super::*;
+    use ::serde::{
+        de::{Error, Visitor},
+        Deserialize, Deserializer, Serialize, Serializer,
+    };
+    use std::fmt::Formatter;
+
+    impl Serialize for GodotString {
+        #[inline]
+        fn serialize<S>(
+            &self,
+            serializer: S,
+        ) -> Result<<S as Serializer>::Ok, <S as Serializer>::Error>
+        where
+            S: Serializer,
+        {
+            serializer.serialize_newtype_struct("GodotString", &*self.to_string())
```

Comment:
Why not just a string? That would simplify `Deserialize` a lot, too.

Reply:
TBH, I don't know why I did it this way. Still learning how to write custom serde impls and for some reason it seemed like a good idea at the time :laughing: